src/TurbulenceClosures/turbulence_closure_implementations/horizontally_curvilinear_anistropic_diffusivity.jl

```diff
@@ -0,0 +1,65 @@
+"""
+    HorizontallyCurvilinearAnisotropicDiffusivity{N, K}
+
+Holds viscosity and diffusivities for models with prescribed isotropic diffusivities.
+"""
+struct HorizontallyCurvilinearAnisotropicDiffusivity{NH, NZ, KH, KZ} <: AbstractTurbulenceClosure
+    νh :: NH
+    νz :: NZ
+    κh :: KH
+    κz :: KZ
+end
+
+"""
+    HorizontallyCurvilinearAnisotropicDiffusivity(; ν=0, κ=0)
+
+Returns parameters for an horizontal diffusivity model on
+curvilinear grids with viscosity `ν` and diffusivities `κ` for each tracer
+field in `tracers`. `ν` and the fields of `κ` may be constants or functions
+of `(x, y, z, t)`, and may represent molecular diffusivities in cases that all flow
+features are explicitly resovled, or turbulent eddy diffusivities that model the effect of
+unresolved, subgrid-scale turbulence.
+`κ` may be a `NamedTuple` with fields corresponding
+to each tracer, or a single number to be a applied to all tracers.
+"""
+function HorizontallyCurvilinearAnisotropicDiffusivity(FT=Float64; νh=0, κh=0, νz=0, κz=0)
+    νh = convert_diffusivity(FT, νh)
+    νz = convert_diffusivity(FT, νz)
+    κh = convert_diffusivity(FT, κh)
+    κz = convert_diffusivity(FT, κz)
+    return HorizontallyCurvilinearAnisotropicDiffusivity(νh, νz, κh, κz)
+end
+
+function with_tracers(tracers, closure::HorizontallyCurvilinearAnisotropicDiffusivity)
+    κh = tracer_diffusivities(tracers, closure.κh)
+    κz = tracer_diffusivities(tracers, closure.κz)
+    return HorizontallyCurvilinearAnisotropicDiffusivity(closure.νh, closure.νz, κh, κz)
+end
+
+calculate_diffusivities!(K, arch, grid, closure::HorizontallyCurvilinearAnisotropicDiffusivity, args...) = nothing
+
+@inline function ∇_κ_∇c(i, j, k, grid, clock, closure::HorizontallyCurvilinearAnisotropicDiffusivity,
+                        c, ::Val{tracer_index}, args...) where tracer_index
+
+    @inbounds κh = closure.κh[tracer_index]
+    @inbounds κz = closure.κz[tracer_index]
+
+    return ∂ⱼκᵢⱼ∂ᵢc(i, j, k, grid, clock, κh, κh, κz, c)
+end
+
+@inline ∂ⱼ_2ν_Σ₁ⱼ(i, j, k, grid, clock, closure::HorizontallyCurvilinearAnisotropicDiffusivity, U, args...) = (
+      δxᶠᵃᵃ(i, j, k, grid, ν_δᶜᶜᶜ, clock, closure.νh, U.u, U.v) / Δxᶠᶜᵃ(i, j, k, grid)
+    - δyᵃᶜᵃ(i, j, k, grid, ν_ζᶠᶠᶜ, clock, closure.νh, U.u, U.v) / Δyᶠᶜᵃ(i, j, k, grid)
```

Comment:
Should this line really start with `-`?

Reply:
This is the sign that shows up in

https://mitgcm.readthedocs.io/en/latest/algorithm/algorithm.html#horizontal-dissipation

On a regular grid / in the continuous equations and with a constant diffusivity this ensure that the contribution of v_xy from the divergence term cancels `v_xy` from the vorticity term (in that case the total diffusivity is u_xx + u_yy as expected).

The sign is reversed in the diffusivity for `v`.

Reply:
Here's a screen shot:

<img width="668" alt="image" src="https://user-images.githubusercontent.com/15271942/108506820-696f8600-7287-11eb-98a6-317fa4008651.png">

Reply:
Thanks!